components/src/dynamo/common/utils/nixl_telemetry.py

```
@@ -0,0 +1,152 @@
+``0`` and binds an ephemeral port, but exposes no accessor for the port it
+actually got, so nothing could name that port as a scrape target.
+
+This module deliberately imports no inference engine: the derivation is pure
+arithmetic over a base port and a rank index, and the callers that know how to
+find a rank index live in the per-backend packages.
+"""
+
+from __future__ import annotations
+
+import os
+from typing import Mapping
```

Ruff (0.16.3)

[warning] Import from `collections.abc` instead: `Mapping`

Import from `collections.abc`

(UP035)